framework/test/mocha/integration/delegates/forge.js

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+'use strict';
+
+const util = require('util');
+const { transfer } = require('@liskhq/lisk-transactions');
+const application = require('../../common/application');
+const Bignum = require('../../../../src/modules/chain/helpers/bignum');
+const random = require('../../common/utils/random');
+const accountFixtures = require('../../fixtures/accounts');
+
+const genesisBlock = __testContext.config.genesisBlock;
+const { NORMALIZER } = global.__testContext.config;
+
+const localCommon = require('../common');
+
+const addTransactionsAndForge = util.promisify(
+	localCommon.addTransactionsAndForge
+);
+const transactionInPool = localCommon.transactionInPool;
+const addTransaction = util.promisify(localCommon.addTransaction);
+const fillPool = util.promisify(localCommon.fillPool);
+const forge = util.promisify(localCommon.forge);
+
+let library;
+
+describe('delegates (forge)', () => {
+	before(done => {
+		application.init(
+			{
+				sandbox: {
+					name: 'lisk_test_delegates_forge',
+				},
+			},
+			(err, scope) => {
+				library = scope;
+				library.modules.blocks.lastBlock.set(genesisBlock);
+				done();
+			}
+		);
+	});
+
+	after(done => {
+		application.cleanup(done);
+	});
+
+	describe('forge', () => {
+		describe('total spending', () => {
+			it('should not include transactions which exceed total spending per account balance', async () => {
+				// Transfer 1LSK to account
+				const account = random.account();
+				const transaction = transfer({
+					amount: new Bignum(NORMALIZER).multipliedBy(1).toString(),
+					recipientId: account.address,
+					passphrase: accountFixtures.genesis.passphrase,
+				});
+				await addTransactionsAndForge(library, [transaction], 0);
+
+				// Transfer 0.5 LSK to a random account - The account balance would be 0.4
+				const trs_06 = transfer({
```

Comment:
Minor thing but the variable name should be camel case

Reply:
We can have more meaningful variable names here, for example, `spend06`, `spend04`, `credit03`